### PR TITLE
Rename fetch_atlas_aal_spm_12 to fetch_atlas_aal

### LINKED ISSUE #764: Function name fetch_atlas_aal_spm_12 is too specific
If we want to update the atlas to a new version, we will break all the existing scripts. We should change it to `fetch_atlas_aal_spm` or `fetch_atlas_aal`.

### PR DESCRIPTION
**Pull request by AlexandreAbraham**
Fix #764

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
@aabadi : do you want to review this?

**Comment by lesteve** [comment]
You need to change the description rst in nilearn/datasets/description/aal_spm_12.rst.

Side remark: it would be nice if we had a way to tests that dataset descriptions were correctly loaded ...

**Comment by aabadie** [comment]
Sorry @GaelVaroquaux, you mispelled my id so I wasn't notified.

**Comment by AlexandreAbraham** [comment]
I rebased, addressed @aabadie's comment, fixed the function that was not working and added tests for all datasets' descriptions.

**Comment by aabadie** [comment]
Thanks for fixing the remaining failing test @AlexandreAbraham. To me all is ok with this PR now (LGTM as you all say ;) ). So I guess it can be merged now : @lesteve @GaelVaroquaux  ?

File: nilearn/datasets/atlas.py [hunk 1]
```diff
@@ -624,17 +636,22 @@ def fetch_atlas_aal_spm_12(data_dir=None, url=None, resume=True, verbose=1):
...
+
     if url is None:
-        baseurl = "http://www.gin.cnrs.fr/AAL_files/aal_for_SPM%i.tar.gz"
-        url = baseurl % spm_version
+        baseurl = "http://www.gin.cnrs.fr/AAL_files/aal_for_%s.tar.gz"
+        url = baseurl % version
     opts = {'uncompress': True}
 
-    dataset_name = "aal_spm_12"
+    dataset_name = "aal_" + version
```
Comment (lesteve) [inline nilearn/datasets/atlas.py]: Stupid question: would the description not be useful irrespective of the SPM version ?

Comment (AlexandreAbraham) [inline nilearn/datasets/atlas.py]: I wonder. The actual file can fit any SPM AAL version but if one day we decide to add an atlas from another source, we may need another file. I have no strong feeling about this ATM.

Comment (lesteve) [inline nilearn/datasets/atlas.py]: Me neither so let's keep it like this for now.

**Comment by lesteve** [comment]
Thanks a lot merging.

**Pull request merged**